dot/rpc/modules/state.go

```diff
@@ -126,11 +126,19 @@ func NewStateModule(net NetworkAPI, storage StorageAPI, core CoreAPI) *StateModu
 // GetPairs returns the keys with prefix, leave empty to get all the keys.
 func (sm *StateModule) GetPairs(r *http.Request, req *[]string, res *[]interface{}) error {
 	// TODO implement change storage trie so that block hash parameter works (See issue #834)
+
+	// Extract the params
 	pReq := *req
 	reqBytes, _ := common.HexToBytes(pReq[0])
+	bhash, _ := common.HexToHash(pReq[1])
+
+	stateRootHash, err := sm.storageAPI.GetStateRootFromBlock(bhash)
+	if err != nil {
+		return err
+	}
 
```

Comment:
I think the block hash parameter is optional (ie. if `len(pReq) == 1` then there is no block hash provided), if no block hash is provided, then the latest data is used which is what was happening before. could you add a case to check `len(pReq)` to see if you should provide a block hash or not?